pebblo/app/service/local_ui/loader_apps.py

```diff
@@ -37,106 +37,139 @@ def __init__(self):
-                        findings_exists = True
-                        findings["findings"] += topic_data["count"]
-                        findings["snippetCount"] += len(topic_data["snippetIds"])
-                        findings["fileCount"] = len(app_data["documents"])
-                        total_snippet_count += findings["snippetCount"]
-                        snippets.extend(
-                            self._get_snippet_details(
-                                topic_data["snippetIds"], app_data["owner"]
-                            )
+        for entity, entity_data in app_data.get("docEntities").items():
+            entity_count += entity_data.get("count")
+            self.loader_findings += entity_data.get("count")
+
+            findings_exists = False
+            for findings in self.loader_findings_list:
+                if findings.get("labelName") == entity:
+                    findings_exists = True
+                    findings["findings"] += entity_data["count"]
+                    findings["snippetCount"] += len(entity_data["snippetIds"])
+                    findings["fileCount"] = len(app_data["documents"])
+                    total_snippet_count += findings["snippetCount"]
+                    snippets.extend(
+                        self._get_snippet_details(
+                            entity_data["snippetIds"], app_data["owner"], entity
                         )
-                        break
-                if not findings_exists:
-                    findings = {
-                        "appName": app_data["name"],
-                        "labelName": topic,
-                        "findings": topic_data["count"],
-                        "findingsType": "topics",
-                        "snippetCount": len(topic_data["snippetIds"]),
-                        "fileCount": len(app_data["documents"]),
-                        "snippets": self._get_snippet_details(
-                            topic_data["snippetIds"], app_data["owner"]
-                        ),
-                    }
+                    )
+                    break
+            if not findings_exists:
+                logger.debug("finding not exist")
+                findings = {
+                    "appName": app_data["name"],
+                    "labelName": entity,
+                    "findings": entity_data["count"],
+                    "findingsType": "entities",
+                    "snippetCount": len(entity_data["snippetIds"]),
+                    "fileCount": len(app_data["documents"]),
+                    "snippets": self._get_snippet_details(
+                        entity_data["snippetIds"], app_data["owner"], entity
+                    ),
+                }
+                total_snippet_count += findings["snippetCount"]
+                shallow_copy = findings.copy()
+                self.loader_findings_list.append(shallow_copy)
+                del findings["snippets"]
```

Comment:
Will snippet be always there any scenario where it will not be present

Reply:
It will be always there, as we are creating `findings` in line no 103